src/client.ts

```diff
@@ -196,19 +197,22 @@ export class MqttClient extends EventEmitter {
 
   private async _destroyClient(force?: boolean) {
     this._clientLogger.trace(`destroying client...`);
-    this.conn.removeAllListeners('error')
+    this.conn.removeAllListeners('error');
+    this.conn.removeAllListeners('close');
+    this.conn.on('close', () => {});

```

Comment:
Why is this needed? I can see why you wanna do this with the `error` event because Node complains when `error` is emitted and there are no listeners. But that's not the case for `close`.